agent/consul/fsm/fsm.go

```diff
@@ -72,12 +73,19 @@ func New(gc *state.TombstoneGC, logOutput io.Writer) (*FSM, error) {
 		return nil, err
 	}
 
+	consulLogger := hclog.New(&hclog.LoggerOptions{
```

Comment:
This feels like a situation where the `New` function should be possibly adjusted to take a `parentLogger hclog.Logger` rather than re-wrap a wrapped logger.

Reply:
So  - I could pass the `logger` that is sitting in Consul Server struct which is an already wrapped logger and make it a child of the Server logger, but because it's wrapped I think I lose the ability to use it as an hclogger (make it a child of a parent)